### PR TITLE
Technical/nijo 1135 move rate checks to separate class

### PR DESCRIPTION
**Pull request by mingazhev**

### CODE REVIEW

**Review by DFSko — APPROVED** [review]

**Review by hasmikharutyunyan — APPROVED** [review]

File: Api/Services/Accommodations/Bookings/BookingRateChecker.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            _settingsService = settingsService;
+            _dateTimeProvider = dateTimeProvider;
+        }
+
+
+        public async Task<Result> Check(AccommodationBookingRequest bookingRequest, BookingAvailabilityInfo availabilityInfo, AgentContext agent)
+        {
+            return await GetSettings()
+                .Ensure(AreAprSettingsSuitable, "You can't book the restricted contract without explicit approval from a Happytravel.com officer.")
+                .Ensure(AreDeadlineSettingsSuitable, "You can't book the contract within deadline without explicit approval from a Happytravel.com officer.");
```
Comment (kirillta) [inline Api/Services/Accommodations/Bookings/BookingRateChecker.cs:27]: Using _Suitable_ here is unclear. _Applicable_ maybe?

File: Api/Services/Accommodations/Bookings/BookingRateChecker.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                {
+                    PassedDeadlineOffersMode.CardAndAccountPurchases => true,
+                    PassedDeadlineOffersMode.CardPurchasesOnly
+                        when bookingRequest.PaymentMethod == PaymentMethods.CreditCard => true,
+                    _ => false
+                };
+            }
+            
+            
+            bool AreAprSettingsSuitable(AccommodationBookingSettings settings)
```
Comment (kirillta) [inline Api/Services/Accommodations/Bookings/BookingRateChecker.cs:49]: Method order is suspicious

**Review by kirillta — APPROVED** [review]

File: Api/Services/Accommodations/Bookings/BookingRateChecker.cs (hunk 2 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Bookings/BookingRateChecker.cs:49]: Will fix in the last PR

File: Api/Services/Accommodations/Bookings/BookingRateChecker.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Bookings/BookingRateChecker.cs:27]: Will fix in the last PR

**Pull request merged**